xen/xenops_interface.ml

```diff
@@ -433,6 +433,8 @@ module Vif = struct
 
 	type id = string * string
 
+	type static_ip_setting = (string * string) list
```

Comment:
Do you need to first string for "enabled"? My idea was to use `Unspecified` to mean "enabled = 0", and `Static` to be "enabled = 1" plus the IP config.

The error may be due to having added the third to xcp-idl, but not installing the updated library before compiling xen-api?